### DISCUSSION TITLE
Event Triggers via Tag Attribute

### DISCUSSION THREAD

**Discussion by RobertJoonas** [plausible/analytics#2237] [Anything Else]
Moving a [docs issue](https://github.com/plausible/docs/issues/275) into a feature request here:

> Just went over your guide, and I was surprised and somewhat disappointed that you don't feature a quick and easy way to track an event via simple attribute addition, for example:
> 
> ```html
> <button data-pa-event="downloadClicked"> Download </button>
> ```
> 
> PS: You could transform the attr name (via npm package) or on-page load if you fear ad-blocking.

**Comment by RobertJoonas** [comment]
We do have something called `tagged-events` planned, which would allow exactly that. This would work with attributes like `data-event-name="Download"` and `data-event-<prop-name>="value"` (for example: `data-event-method="Magnet"`). This could be implemented for links and forms, but I'm not sure about buttons. The thing is that buttons don't have any standard default behavior, which we need to know in order to intercept possible navigation. Say a button is clicked that is supposed to:

1) send a custom event to Plausible
2) take the user to another page on your site

As navigation can be requested with the button click, the Plausible request is not guaranteed to run to completion. In order to guarantee it, we'd need to delay navigation until the Plausible request is sent.

Hope this makes sense.

**Reply by MentalGear** [reply]
How's about putting your analytics event listener in front of all others attached, and using sendBeacon for all your calls?

**Reply by RobertJoonas** [reply]
Agreed that sendBeacon would be the best solution. The only con is no support for Internet Explorer which we still want to keep up in the existing script. Our long-term plan is to build a version 2 tracker script, that would only use Beacon for sending event requests. This should make it much easier to implement custom events too (not having to worry about navigation). Unfortunately, I cannot make any timeline promises on tracker v2 at this stage. Thanks for understanding!

**Reply by alexanderkirtzel** [reply]
Bringing the tracking implementation directly to your markup instead of hiding it somewhere in your code has lots of advantages. But it’s not just limited to a click. And you might want to collect additional parameters as props to send them to Plausible.
I’m working on an open and vendor-agnostic approach to set up analytics implementations directly in your markup that way and recently added a Plausible destination.

```html
<button data-elb=“download” data-elbaction=“click:clicked” data-elb-download=“file:name.pdf”> Download </button>
```

Would love to get your feedback on it, @MentalGear

**Reply by MentalGear** [reply]
Thank you for the hint, Alex! It looks very interesting. If I get the chance to try it out, I'll let you know more.
On a small note: Took a quick look at the docs, and it seems on github the "pulse" event exist, while on the website it doesn't.

**Reply by alexanderkirtzel** [reply]
Awesome! But since this is also about Plausible here, my suggestion would be to introduce tags like `data-pa-prop="key:value"` or `data-pa-prop-<key>="value"` directly on an element.
And thanks for the hint, will sync it!

**Reply by MentalGear** [reply]
If you're concerned about IE support, why not just do a feature detection for beacon, and use a fallback for IE ?
Thinking about it, you might lose more data by not using beacon than most audiences have IE users. Seems non logical.

**Reply by RobertJoonas** [reply]
> why not just do a feature detection for beacon, and use a fallback for IE?

We don't want to mix Beacon with XMLHttpRequest in one script, as this will make the code more complex than it hasto be. That's why we're planning to keep that script around as a legacy option and start from a clean slate with Beacon at some point in the future. Hope this makes sense!
